File: lavcode-node/src/actions/folder/dtos/update-folder.dto.ts
import { V } from "@ipare/validator";
import { UpsertIconDto } from "../../../dtos/upsert-icon.dto";

export class UpdateFolderDto {
  @V().Required().IsString()
  name!: string;

  @V().Required().IsNumber()
  order!: number;

  @V().Description("空则不修改图标")
  icon!: UpsertIconDto;
}

File: Lavcode.Asp/Dtos/UpdateFolderDto.cs
using System.ComponentModel.DataAnnotations;

namespace Lavcode.Asp.Dtos
{
    public class UpdateFolderDto
    {
        [Required]
        public string Name { get; set; } = null!;

        [Required]
        public int Order { get; set; }

        /// <summary>
        /// 空则不修改图标
        /// </summary>
        [Required]
        public UpsertIconDto? Icon { get; set; }
    }
}
